util/test/testutil.go

```diff
@@ -107,6 +108,30 @@ rawN1NFkKr5MCxl5Q4veocU94ThOlFdYgreyVX6s40ZL1eF0RvAQ+e0oFT7SfCHu
 B3XwyYtAFsaO5r7oEc1Bv6oNSbE+FNJzRdjkWEIhdLVKlepil/w=
 -----END RSA PRIVATE KEY-----`)
 
+type JSONWebKeySet struct {
+	Keys []JSONWebKey `json:"keys"`
+}
+
+type JSONWebKey struct {
+	// Cryptographic key, can be a symmetric or asymmetric key.
+	Key interface{}
+	// Key identifier, parsed from `kid` header.
+	KeyID string
+	// Key algorithm, parsed from `alg` header.
+	Algorithm string
+	// Key use, parsed from `use` header.
+	Use string
+
+	// X.509 certificate chain, parsed from `x5c` header.
+	Certificates []*x509.Certificate
+	// X.509 certificate URL, parsed from `x5u` header.
+	CertificatesURL *url.URL
+	// X.509 certificate thumbprint (SHA-1), parsed from `x5t` header.
+	CertificateThumbprintSHA1 []byte
+	// X.509 certificate thumbprint (SHA-256), parsed from `x5t#S256` header.
+	CertificateThumbprintSHA256 []byte
+}
```

Comment:
Are these just copy/pasted from go-jose? If so, wanna add a comment linking to the original source and explaining why we copied them?

Reply:
Just to stay 100% in the clear on the Apache license, I'd also drop the code comments or replace them with your own if those are copy/pasted.

Reply:
This package go-jose/go-jose/v3 is not used much. You need to replace v3 or copy the structure.